src/ai/formula/ai.hpp

```diff
@@ -36,12 +35,10 @@
 #include <vector>                       // for vector
 
 namespace ai { class ai_context; }
-namespace pathfind { struct plain_route; }  // lines 57-57
 struct map_location;
 
 namespace wfl {
 	struct formula_error;
-	class variant;
```

Comment:
I'm not sure if it's in scope for this PR, but these two namespace declarations could be merged into one.

Reply:
I’m not quite sure what to do, so I won’t proceed with the merger.

Reply:
Avoiding any complex changes is good in a PR that touches this many files. It's much easier to review a change that is purely deleting self-contained lines, or self-contained sets of lines.